src/main/java/org/broadinstitute/hellbender/tools/walkers/qc/CheckPileup.java

```diff
@@ -0,0 +1,226 @@
+ *  </ul>
+ * </p>
+ * <h4>Example of consensus pileup for SNP or non-variant sites</h4>
+ * <pre>
+ *     seq1  60  T  T  66  0  99  13  ...........^~.^~.   9<<55<;<<<<<<
+ *     seq1  61  G  G  72  0  99  15  .............^~.^y. (;975&;<<<<<<<<
+ *     seq1  62  T  T  72  0  99  15  .$..............    <;;,55;<<<<<<<<
+ *     seq1  63  G  G  72  0  99  15  .$.............^~.  4;2;<7:+<<<<<<<
+ *     seq1  64  G  G  69  0  99  14  ..............  9+5<;;;<<<<<<<
+ *     seq1  65  A  A  69  0  99  14  .$............. <5-2<;;<<<<<<;
+ *     seq1  66  C  C  66  0  99  13  .............   &*<;;<<<<<<8<
+ *     seq1  67  C  C  69  0  99  14  .............^~.    ,75<.4<<<<<-<<
+ *     seq1  68  C  C  69  0  99  14  ..............  576<;7<<<<<8<< *
+ * </pre>
+ *
+ * <h4>Example of consensus pileup for indels</h4>
+ * <pre>
+ *     Escherichia_coli_K12	3995037	*	*\/*	430	0	37	144	*	+A	143	1	0
+ *     Escherichia_coli_K12	3995279	*	*\/*	202	0	36	68	*	+A	67	1	0
+ *     Escherichia_coli_K12	3995281	*	*\/*	239	0	36	67	*	-CG	66	1	0
+ * </pre>
+ * <p>
+ *     See <a href="http://samtools.sourceforge.net/cns0.shtml/">Consensus pileup format (deprecated)</a> for more details.
+ * </p>
+ *
+ * <h3>Input</h3>
+ * <p>A BAM file containing your aligned sequence data and a pileup file generated by Samtools covering the region you
+ * want to examine.</p>
+ *
+ * <h3>Output</h3>
+ * <p>A text file listing mismatches between the input pileup and the GATK's internal pileup. If there are no mismatches, the output file is empty.</p>
+ *
+ * <h3>Usage example</h3>
+ * <pre>
+ * ./gatk-launch CheckPileup \
+ *   -R reference.fasta \
+ *   -I your_data.bam \
+ *   --pileup pileup_file.txt \
+ *   -L chr1:257-275 \
+ *   -O output_file_name
+ * </pre>
+ */
+@CommandLineProgramProperties(
+    summary = "At every locus in the input set, compares the pileup data (reference base, aligned base from each overlapping"
+        + "read, and quality score) generated internally by GATK to a reference pileup data generated by Samtools. Note "
+        + "that the pileup program has been replaced in Samtools by mpileup, which produces a slightly different output "
+        + "format by default.",
+    oneLineSummary = "Compare GATK's internal pileup to a reference Samtools pileup",
+    programGroup = QCProgramGroup.class)
+public class CheckPileup extends LocusWalker {
+
+    /**
+     * This is the existing pileup against which we'll compare GATK's internal pileup at each genome position in the desired interval.
+     */
+    @Argument(fullName = "pileup", shortName = "pileup", doc="Pileup generated by Samtools")
+    FeatureInput<SAMPileupFeature> pileup;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "An output file created by the walker. Will overwrite contents if file exists")
+    private PrintStream out;
```

Comment:
Huh, I didn't realize PrintStream had a string constructor.  This saves a few lines, but it's probably better use a `File` here and manually instantiate the stream in order to be able to throw a clearer `UserException` when an invalid path is given.  Right now it will result in something like 
`A USER ERROR has occurred: Invalid command line: Argument  output has a bad value: /some/invalid/path. Problem constructing PrintStream from the string '/some/invalid/path'.` 

Which isn't that bad, but would be clearer as a `CouldNotCreateOutputFile`  Not a big deal either way.  There's a long neglected plan to make generating and cleaning up output streams cleaner, but it hasn't happened yet.